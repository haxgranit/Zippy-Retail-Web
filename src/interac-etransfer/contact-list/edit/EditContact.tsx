import { useState } from 'react';
import {
  Col, Row, Form, Button,
} from 'react-bootstrap';
import { useLocation } from 'react-router-dom';
import StepComponent from '../../../common/StepComponent';
import CommonHeader from '../../../common/CommonHeader';
import IdentityVerificationModal from './IdentityVerificationModal';
import { formatContactName } from '../../../Helpers';

const Divider = () => <div className="border-top my-3" />;
const HorizontalLine = () => (
  <hr
    style={{
      height: '2px',
      borderWidth: '0px',
      color: 'gray',
      background: 'gray',
      marginTop: 40,
    }}
  />
);

const LeftCol = ({ currentStep, setCurrentStep, setPageIndex }: any) => (
  <Col xs={9}>
    <Row>
      <Col>
        <StepComponent
          steps={2}
          currentStep={currentStep}
          setCurrentStep={setCurrentStep}
          setPageIndex={setPageIndex}
        />
      </Col>
    </Row>
    <Divider />
    <Row>
      <Col>
        <h3>Contact Profile:</h3>
        <ul style={{ marginTop: 50 }}>
          <li>
            Change the contact profile by deleting or typing over the existing
            content then selecting Next
          </li>
          <li>
            Enter an email address or mobile phone number for your contact
          </li>
        </ul>
      </Col>
    </Row>
  </Col>
);

const RightCol = () => (
  <Col xs={3}>
    <div className="border p-2 rounded">
      <b>You can also:</b>
      <br />
      View upcoming bill payments and transfers &gt;
      <br />
      Stop a payment &gt;
      <br />
      Change your statement preferences &gt;
    </div>
  </Col>
);

const EditContact: React.FC = () => {
  const { state } = useLocation();
  const selectedContact = state?.item;
  const [show, setShow] = useState(false);

  return (
    <>
      <div>
        <IdentityVerificationModal
          show={show}
          handleClose={() => setShow(false)}
          selectedContact={selectedContact}
        />
        <Row>
          <Col>
            <CommonHeader title="EDIT CONTACT" print={false} />
          </Col>
        </Row>
        <Row>
          <LeftCol currentStep={1} setCurrentStep={() => {}} />
          <RightCol />
        </Row>
        <Row style={{ marginTop: 100 }}>
          <Col xs="4" style={{ display: 'flex', alignItems: 'center' }}>
            <span style={{ display: 'inline-block', width: 'fit-content' }}>
              Name:
            </span>
          </Col>
          <Col xs="4" style={{ display: 'flex', alignItems: 'center' }}>
            <Form.Control
              type="text"
              value={formatContactName(selectedContact?.firstName, selectedContact?.lastName)}
              style={{
                display: 'inline-block',
                width: 250,
                marginLeft: 10,
                marginRight: 10,
              }}
            />
          </Col>
          <Col xs="4" style={{ display: 'flex', alignItems: 'center' }}>
            {}
          </Col>
        </Row>
        <HorizontalLine />
        <Row style={{ marginTop: 30 }}>
          <Col>
            <span>Details</span>
          </Col>
          <Col>
            <div className="d-flex">
              <div
                style={{
                  width: 20,
                  height: 25,
                  border: '1px dotted grey',
                  textAlign: 'center',
                  marginRight: 10,
                  marginLeft: 10,
                }}
              >
                P
              </div>
              <a href="/">Mobile phone</a>
            </div>
          </Col>
          <Col>{}</Col>
        </Row>
        <HorizontalLine />
        <Row style={{ marginTop: 30 }}>
          <Col xs="4">
            <span style={{ display: 'flex', alignItems: 'center' }}>
              Email Address:
            </span>
          </Col>
          <Col>
            <Form.Control
              type="text"
              value={selectedContact?.email}
              style={{
                width: 250,
                marginLeft: 10,
                marginRight: 10,
              }}
            />
          </Col>
          <Col
            style={{
              display: 'flex',
              alignItems: 'center',
              justifyContent: 'left',
            }}
          >
            <a href="/">Delete</a>
          </Col>
        </Row>
        <HorizontalLine />
        <Row style={{ marginTop: 30 }}>
          <Col>
            <span>Notification Language:</span>
          </Col>
          <Col>
            <Form.Select
              style={{
                width: 250,
                marginLeft: 10,
                marginRight: 10,
                marginBottom: 50,
              }}
            >
              <option>English</option>
            </Form.Select>
          </Col>
          <Col>{}</Col>
        </Row>
        <HorizontalLine />
        <Row>
          <Col style={{ padding: 0, background: 'white' }}>
            <Button
              style={{ margin: 0, background: 'white', border: 'none' }}
              variant="light"
              className="d-flex"
            >
              <div
                style={{
                  width: 20,
                  height: 25,
                  border: '1px dotted grey',
                  textAlign: 'center',
                  marginRight: 10,
                }}
              >
                P
              </div>
              Cancel
            </Button>
          </Col>
          <Col style={{ display: 'flex', justifyContent: 'flex-end' }}>
            <Button
              className="btn btn-danger"
              onClick={() => setShow(true)}
              style={{ marginLeft: 15 }}
            >
              Next
            </Button>
          </Col>
        </Row>
        <HorizontalLine />
        <Row>
          <div className="mt-4">
            <div className="d-flex">
              <div
                style={{
                  width: 20,
                  height: 25,
                  border: '1px dotted grey',
                  textAlign: 'center',
                  marginRight: 10,
                }}
              >
                P
              </div>
              <span style={{ fontWeight: 'bold' }}>Note:</span>
            </div>
            <p>
              <span>Your use of Interac e-Transfer is subject to the </span>
              <Row>
                <Col md={12} className="d-flex align-items-center mt-2">
                  <div
                    style={{
                      width: 20,
                      height: 25,
                      border: '1px dotted grey',
                      textAlign: 'center',
                      marginRight: 10,
                    }}
                  >
                    P
                  </div>
                  <span>
                    Interac e-Transfer Terms and Conditions (PDF, 197KB).
                  </span>
                </Col>
              </Row>
            </p>
          </div>
        </Row>
      </div>
    </>
  );
};

export default EditContact;
