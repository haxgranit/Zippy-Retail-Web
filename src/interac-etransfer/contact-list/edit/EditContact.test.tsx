import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import EditContact from './EditContact';

// Configure enzyme for react 17
Enzyme.configure({ adapter: new Adapter() });

const CONTACT_MOCK = {
  key: 1,
  name: '392 Jones',
  lang: 'English',
  email: 'lu_ben2002@yahoo.com',
  phone: '',
};

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useLocation: () => ({
    item: CONTACT_MOCK,
  }),
}));

describe('EditContact Component', () => {
  it('should render title', () => {
    const setShow = jest.fn();
    React.useState = jest
      .fn()
      .mockImplementationOnce((x) => [x, setShow]);
    const wrapper = shallow(<EditContact />);
    const header = wrapper.find('CommonHeader');
    expect(header).toHaveLength(1);
  });

  it('click next button', () => {
    const setShow = jest.fn();
    React.useState = jest
      .fn()
      .mockImplementationOnce((x) => [x, setShow]);
    const wrapper = shallow(<EditContact />);
    wrapper.find('Button').at(1).simulate('click');
    expect(setShow).toHaveBeenCalled();
  });

  it('click close modal button on IdentityVerificationModal', () => {
    const setShow = jest.fn();
    const setStep = jest.fn();
    React.useState = jest
      .fn()
      .mockImplementationOnce((x) => [x, setShow])
      .mockImplementationOnce((x) => [x, setStep]);
    const wrapper = shallow(<EditContact />);
    wrapper.find('Button').at(1).simulate('click');
    expect(setShow).toHaveBeenCalled();
    const modalWrapper = wrapper.find('IdentityVerificationModal').dive();
    const firstStepWrapper = modalWrapper.find('VerificationStep').dive();
    firstStepWrapper.find('Button[variant="link"]').at(0).simulate('click');
    firstStepWrapper.find('button').at(0).simulate('click');
    firstStepWrapper.find('button').at(1).simulate('click');
    expect(setShow).toHaveBeenCalled();
  })
});
