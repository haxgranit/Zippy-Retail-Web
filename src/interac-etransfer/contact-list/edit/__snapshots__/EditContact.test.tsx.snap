// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditContact Component should render title 1`] = `
<div>
  <div
    class="d-flex justify-content-between align-items-center"
  >
    <h3>
      Edit Contact
    </h3>
  </div>
  <div
    class="content-wrapper"
  >
    <div
      class="row"
    >
      <div
        class="col-9"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="d-flex align-items-center mt-3 mb-3"
            >
              <span
                style="font-weight: bold; margin-right: 10px;"
              >
                Steps: 
              </span>
              <div
                role="button"
                style="width: 25px; height: 25px; border-radius: 50px; border: 2px solid #c41f3e; display: flex; justify-content: center; align-items: center; background-color: rgb(196, 31, 62); color: white;"
              >
                1
              </div>
              <div
                style="height: 2px; width: 15px; background-color: grey;"
              />
              <div
                role="button"
                style="width: 25px; height: 25px; border-radius: 50px; border: 2px solid grey; display: flex; justify-content: center; align-items: center;"
              >
                2
              </div>
            </div>
          </div>
        </div>
        <div
          class="border-top my-3"
        />
        <div
          class="row"
        >
          <div
            class="col"
          >
            <h3>
              Contact Profile:
            </h3>
            <ul
              style="margin-top: 50px;"
            >
              <li>
                Change the contact profile by deleting or typing over the existing content then selecting Next
              </li>
              <li>
                Enter an email address or mobile phone number for your contact
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="col-3"
      >
        <div
          class="border p-2 rounded"
        >
          <b>
            You can also:
          </b>
          <br />
          View upcoming bill payments and transfers &gt;
          <br />
          Stop a payment &gt;
          <br />
          Change your statement preferences &gt;
        </div>
      </div>
    </div>
    <div
      class="row"
      style="margin-top: 100px;"
    >
      <div
        class="col-4"
        style="display: flex; align-items: center;"
      >
        <span
          style="display: inline-block;"
        >
          Name:
        </span>
      </div>
      <div
        class="col-4"
        style="display: flex; align-items: center;"
      >
        <input
          class="form-control"
          style="display: inline-block; width: 250px; margin-left: 10px; margin-right: 10px;"
          type="text"
          value=""
        />
      </div>
      <div
        class="col-4"
        style="display: flex; align-items: center;"
      />
    </div>
    <hr
      style="height: 2px; border-width: 0px; color: gray; background: gray; margin-top: 40px;"
    />
    <div
      class="row"
      style="margin-top: 30px;"
    >
      <div
        class="col"
      >
        <span>
          Details
        </span>
      </div>
      <div
        class="col"
      >
        <div
          class="d-flex"
        >
          <div
            style="width: 20px; height: 25px; border: 1px dotted grey; text-align: center; margin-right: 10px; margin-left: 10px;"
          >
            P
          </div>
          <a
            href="/"
          >
            Mobile phone
          </a>
        </div>
      </div>
      <div
        class="col"
      />
    </div>
    <hr
      style="height: 2px; border-width: 0px; color: gray; background: gray; margin-top: 40px;"
    />
    <div
      class="row"
      style="margin-top: 30px;"
    >
      <div
        class="col-4"
      >
        <span
          style="display: flex; align-items: center;"
        >
          Email Address:
        </span>
      </div>
      <div
        class="col"
      >
        <input
          class="form-control"
          style="width: 250px; margin-left: 10px; margin-right: 10px;"
          type="text"
          value=""
        />
      </div>
      <div
        class="col"
        style="display: flex; align-items: center; justify-content: left;"
      >
        <a
          href="/"
        >
          Delete
        </a>
      </div>
    </div>
    <hr
      style="height: 2px; border-width: 0px; color: gray; background: gray; margin-top: 40px;"
    />
    <div
      class="row"
      style="margin-top: 30px;"
    >
      <div
        class="col"
      >
        <span>
          Notification Language:
        </span>
      </div>
      <div
        class="col"
      >
        <select
          class="form-select"
          style="width: 250px; margin-left: 10px; margin-right: 10px; margin-bottom: 50px;"
        >
          <option>
            English
          </option>
        </select>
      </div>
      <div
        class="col"
      />
    </div>
    <hr
      style="height: 2px; border-width: 0px; color: gray; background: gray; margin-top: 40px;"
    />
    <div
      class="row"
    >
      <div
        class="col"
        style="padding: 0px; background: white;"
      >
        <button
          class="d-flex btn btn-light"
          style="margin: 0px; background: white;"
          type="button"
        >
          <div
            style="width: 20px; height: 25px; border: 1px dotted grey; text-align: center; margin-right: 10px;"
          >
            P
          </div>
          Cancel
        </button>
      </div>
      <div
        class="col"
        style="display: flex; justify-content: flex-end;"
      >
        <button
          class="btn zippy-btn btn btn-primary"
          style="margin-left: 15px;"
          type="button"
        >
          Next
        </button>
      </div>
    </div>
    <hr
      style="height: 2px; border-width: 0px; color: gray; background: gray; margin-top: 40px;"
    />
    <div
      class="row"
    >
      <div
        class="mt-4"
      >
        <div
          class="d-flex"
        >
          <div
            style="width: 20px; height: 25px; border: 1px dotted grey; text-align: center; margin-right: 10px;"
          >
            P
          </div>
          <span
            style="font-weight: bold;"
          >
            Note:
          </span>
        </div>
        <div>
          <span>
            Your use of Interac e-Transfer is subject to the 
          </span>
          <div
            class="row"
          >
            <div
              class="d-flex align-items-center mt-2 col-md-12"
            >
              <div
                style="width: 20px; height: 25px; border: 1px dotted grey; text-align: center; margin-right: 10px;"
              >
                P
              </div>
              <span>
                Interac e-Transfer Terms and Conditions (PDF, 197KB).
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="footer-content-wrapper"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-12"
        >
          Quick Links
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm-2"
        >
          <ul>
            <li>
              <a
                class="text-decoration-none"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              Please Review Our Legal Stuff by visiting our
               
              <a
                class="text-decoration-none"
                href="/legal"
              >
                Privacy Policy | TOU | EULA | CardHolder T&C Pages
              </a>
            </li>
          </ul>
        </div>
        <div
          class="col-sm-2"
        >
          8465 W Sahara Ave. Las Vegas, NEVADA
          <ul>
            <li>
              <a
                href="tel:+17026231876"
              >
                1.702.623.1876
              </a>
            </li>
            <li>
              <a
                href="mailto:contact@zippy.cash"
              >
                contact@zippy.cash
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-12 col-sm-12 col-xs-12 text-left"
        >
          © 2021 Zippy Cash LLC  All rights reserved.
          <br />
          <br />
          The Program Manager for the Zippy Cash® Prepaid Visa Debit Card is PayQuicker, LLC
          <br />
          Zippy Cash® is a registered trademark of Zippy Cash, LLC
          <br />
          The Zippy Cash Prepaid Visa Debit Card is issued by Metropolitan Commercial Bank (Member FDIC) pursuant to a license from Visa U.S.A. Inc. “Metropolitan Commercial Bank” and “Metropolitan” are registered trademarks of Metropolitan Commercial Bank ©2014. BY USING THIS CARD YOU AGREE TO THE TERMS AND CONDITIONS OF THE CARDHOLDER AGREEMENT AND FEE SCHEDULE, IF ANY. All trademarks and service marks are the property of their respective owners.
          <br />
          In compliance with the USA PATRIOT Act* and other laws, anyone who is a US resident over 18 with a valid Social Security number and a verifiable physical US address (no PO boxes or APO addresses) can apply for an Zippy Cash Card. This information must be provided, along with other information pertinent to receiving approval, when applying for a Zippy Cash Card. Currently, we do not issue the Zippy Cash Card to residents of Puerto Rico, Guam, the Virgin Islands, American Samoa, or other US territories.
          <br />
          *Important information for opening a card account: To help the federal government fight the funding of terrorism and money laundering activities, the USA PATRIOT Act requires all financial institutions and their third parties to obtain, verify and record information that identifies each person who opens a card account. What this means for you: When you open a card account, we will ask for your name, address, date of birth, social security number or country identification number, and other information that will allows us to identify you. We may also request a copy of your driver’s license or other documentation bearing your photo as verification of your identity.
          <br />
          Apple and the Apple logo are trademarks of Apple Inc., registered in the U.S. and other countries. App Store is a service mark of Apple Inc. Google Play is a trademark of Google Inc. Data rates may apply. Samsung Pay is a trademark of Samsung Inc.
        </div>
      </div>
    </div>
    <footer
      class="container mb-3"
    >
      <div
        class="d-none d-md-block"
      >
        <div
          class="d-flex justify-content-between border-bottom"
        />
        <div
          class="d-flex justify-content-between border-top"
        >
          <div
            class=""
          >
            <div
              class="footer-link mt-1"
            >
              <a
                class="text-decoration-none"
                href="/personal-profile"
              >
                dev docs
              </a>
            </div>
            <div
              class="footer-link mt-1"
            >
              <a
                class="text-decoration-none"
                href="/"
              >
                Lore
              </a>
            </div>
            <div
              class="footer-link mt-1"
            >
              <a
                class="text-decoration-none"
                href="/"
              >
                Ipsum
              </a>
            </div>
            <div
              class="footer-link mt-1"
            >
              <a
                class="text-decoration-none"
                href="/"
              >
                Nullam Atoidio
              </a>
            </div>
          </div>
          <div
            class="copyright mt-1"
          >
            <span>
              © 2021 Lorem ipsum dolor sit amet, consectetuer
            </span>
          </div>
        </div>
      </div>
      <div
        class="d-block d-md-none"
      >
        <div
          class="d-flex border-bottom"
        >
          <div
            class="d-flex flex-column me-2"
          >
            <div
              class="footer-link mt-1"
            >
              <a
                class="text-decoration-none"
                href="/personal-profile"
              >
                dev docs
              </a>
            </div>
            <div
              class="footer-link mb-1"
            >
              <a
                class="text-decoration-none"
                href="/"
              >
                Lore
              </a>
            </div>
            <div
              class="footer-link mb-1"
            >
              <a
                class="text-decoration-none"
                href="/"
              >
                Ipsum
              </a>
            </div>
            <div
              class="footer-link mb-1"
            >
              <a
                class="text-decoration-none"
                href="/"
              >
                Nullam Atoidio
              </a>
            </div>
          </div>
        </div>
        <div
          class="d-flex justify-content-end border-top"
        >
          <div
            class="copyright mt-2 text-end small"
          >
            <span>
              © 2021 Lorem ipsum dolor sit amet, consectetuer
            </span>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
